diff --git a/include/spdlog/tweakme.h b/include/spdlog/tweakme.h
index 31e417ab..543035cc 100644
--- a/include/spdlog/tweakme.h
+++ b/include/spdlog/tweakme.h
@@ -97,7 +97,7 @@
 // Uncomment to customize short level names (e.g. "MT")
 // These can be longer than one character.
 //
-// #define SPDLOG_SHORT_LEVEL_NAMES { "T", "D", "I", "W", "E", "C", "O" }
+#define SPDLOG_LEVEL_NAMES = {"TRACE","DEBUG","INFO","WARNING","ERROR","CRITICAL","OFF"}
 ///////////////////////////////////////////////////////////////////////////////
 
 ///////////////////////////////////////////////////////////////////////////////
