diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
index 5fa34150a..f83ca12fc 100644
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -1547,7 +1547,7 @@ config("chromium_code") {
   } else {
     cflags = [ "-Wall" ]
     if (treat_warnings_as_errors) {
-      cflags += [ "-Werror" ]
+      cflags += [ ]
 
       # The compiler driver can sometimes (rarely) emit warnings before calling
       # the actual linker.  Make sure these warnings are treated as errors as

