diff -Naur libmemcached.orig/tests/mem_functions.c libmemcached/tests/mem_functions.c
--- libmemcached.orig/tests/mem_functions.c	2010-08-02 09:48:42.000000000 -0700
+++ libmemcached/tests/mem_functions.c	2011-03-07 08:41:47.757192727 -0800
@@ -5437,7 +5437,7 @@
   memcached_server_instance_st instance_one;
   memcached_server_instance_st instance_two;
 
-  if (memcached_server_count(memc) < 3 || pre_replication(memc) != MEMCACHED_SUCCESS)
+  if (memcached_server_count(memc) < 3 || pre_replication(memc) != TEST_SUCCESS)
     return TEST_SKIPPED;
 
   memcached_return_t rc;
