diff --git a/build/fbcode_builder/CMake/FindGlog.cmake b/build/fbcode_builder/CMake/FindGlog.cmake
index a589b2e37..752647cb3 100644
--- a/build/fbcode_builder/CMake/FindGlog.cmake
+++ b/build/fbcode_builder/CMake/FindGlog.cmake
@@ -7,13 +7,18 @@
 # GLOG_LIBRARIES - link these to use Glog

 include(FindPackageHandleStandardArgs)
+include(SelectLibraryConfigurations)

-find_library(GLOG_LIBRARY glog
+find_library(GLOG_LIBRARY_RELEASE glog
+  PATHS ${GLOG_LIBRARYDIR})
+find_library(GLOG_LIBRARY_DEBUG glogd
   PATHS ${GLOG_LIBRARYDIR})

 find_path(GLOG_INCLUDE_DIR glog/logging.h
   PATHS ${GLOG_INCLUDEDIR})

+select_library_configurations(GLOG)
+
 find_package_handle_standard_args(glog DEFAULT_MSG
   GLOG_LIBRARY
   GLOG_INCLUDE_DIR)
