diff --git a/toolchain/gcc_toolchain.gni b/toolchain/gcc_toolchain.gni
index e7c88708c..c57642d38 100644
--- a/toolchain/gcc_toolchain.gni
+++ b/toolchain/gcc_toolchain.gni
@@ -607,6 +607,7 @@ template("clang_toolchain") {
                            [
                              "strip",
                              "is_clang_analysis_supported",
+                             "extra_ldflags",
                              "enable_linker_map",
                              "use_unstripped_as_runtime_outputs",
                            ])
diff --git a/toolchain/linux/BUILD.gn b/toolchain/linux/BUILD.gn
index 1b0f22059..924225cfe 100644
--- a/toolchain/linux/BUILD.gn
+++ b/toolchain/linux/BUILD.gn
@@ -111,6 +111,8 @@ gcc_toolchain("x86") {
 }

 clang_toolchain("clang_x64") {
+  extra_ldflags = "-Wl,-rpath=\\\$ORIGIN -Wl,--enable-new-dtags"
+
   # Output linker map files for binary size analysis.
   enable_linker_map = true

